Callisto/PostSlackAction.swift

```diff
@@ -0,0 +1,118 @@
+
+        let infos = inputFiles.map { BuildInformation.read(url: $0) }.compactMap { result -> BuildInformation? in
+            switch result {
+            case .success(let info):
+                return info
+            case .failure(let error):
+                LogError("\(error)")
+                return nil
+            }
+        }
+
+        if infos.allSatisfy({ $0.isEmpty }) {
+            LogMessage("No Build issues found")
+            quit(.success)
+        }
+
+        let branch = self.currentBranch()
+        let slackMessage = self.makeSlackMessage(for: branch, infos: infos)
+        guard let data = slackMessage.jsonDataRepresentation() else { quit(.jsonConversationFailed) }
+
+        self.slackController.post(data: data)
+        quit(.success)
+    }
+}
+
+// MARK: - Private
+
+private extension PostSlackAction {
+
+    func currentBranch() -> Branch {
+        switch self.githubController.branch(named: defaults.branch) {
+        case .success(let branch):
+            return branch
+        case .failure(let error):
+            LogError(error.localizedDescription)
+            quit(.reloadBranchFailed)
+        }
+    }
+
+    func makeSlackMessage(for branch: Branch, infos: [BuildInformation]) -> SlackMessage {
+        let message = SlackMessage()
+
+        // Overview
+        let overViewAttachment = SlackAttachment(type: .good)
+        overViewAttachment.title = branch.title
+        overViewAttachment.titleURL = branch.url
+        overViewAttachment.footer = "Ignored: \(self.ignore.joined(separator: ", "))"
+        message.add(attachment: overViewAttachment)
+
+        // Errors
+        let errorAttachments = SlackAttachment(type: .danger)
+        let errors = infos.flatMap { $0.errors }
+        for compilerMessage in errors {
+            if self.ignore.contains(where: { compilerMessage.description.contains($0)}) { continue }
+
+            errorAttachments.addField(SlackField(message: compilerMessage))
+        }
+        message.add(attachment: errorAttachments)
+
+        // Warnings
+        let warningAttachments = SlackAttachment(type: .warning)
+        let warnings = infos.flatMap{ $0.warnings }
+        for compilerMessage in warnings {
+            if self.ignore.contains(where: { compilerMessage.description.contains($0)}) { continue }
+
+            warningAttachments.addField(SlackField(message: compilerMessage))
+        }
+        message.add(attachment: warningAttachments)
+
+        // Unit Tests
+        let unitTestAttachments = SlackAttachment(type: .danger)
+        unitTestAttachments.colorHex = "blue"
+        let unitTests = infos.flatMap { $0.unitTests }
+        for compilerMessage in unitTests {
+            if self.ignore.contains(where: { compilerMessage.description.contains($0)}) { continue }
+
+            unitTestAttachments.addField(SlackField(message: compilerMessage))
+        }
+        message.add(attachment: unitTestAttachments)
```

Comment:
nit: those 3 blocks look quite repetitive, could be factored out into separate func which takes `[String]` and `SlackAttachment` as parameter

Reply:
I changed it but I don't think it is less code or more readable now